src/salaryAndBenefits/SalaryAndBenefits.tsx

```diff
@@ -1,16 +1,72 @@
+"use client"
+
 import styles from "./salaryAndBenefits.module.css";
 import Text from "src/components/text/Text";
 import {SalaryAndBenefits as SalaryAndBenefitsPayload} from "studio/lib/payloads/salaryAndBenefits";
 import Benefit from './components/benefit/Benefit';
+import { useState } from "react";
+import InputField from "src/components/forms/inputField/InputField";
+import { RadioButtonGroup } from "src/components/forms/radioButtonGroup/RadioButtonGroup";
+import { calculatePension, calculateSalary } from "src/salaryAndBenefits/utils/calculateSalary";
+
+interface IOption {
+  id: string;
+  label: string;
+  disabled?: boolean;
+  currentChecked: boolean;
+}
+
+const options: IOption[] = [
+  { id: "bachelor", label: "Bachelor", currentChecked: false, disabled: false },
+  { id: "master", label: "Master", currentChecked: true, disabled: false },
+];
+
 
 interface SalaryAndBenefitsProps {
   salaryAndBenefits: SalaryAndBenefitsPayload
 }
 
 const SalaryAndBenefits = ({salaryAndBenefits}: SalaryAndBenefitsProps) => {
+  const thisYear = new Date().getFullYear();
+  const [examinationYear, setExaminationYear] = useState(thisYear);
+  const [selectedOption, setSelectedOption] = useState("master");
+  const [salary, setSalary] = useState(
+    calculateSalary(thisYear, examinationYear, selectedOption)
+  );
+
+  const handleChange = (name: string, value: string) => {
+    if (name === "examinationYear") {
+      setExaminationYear(parseInt(value));
+      setSalary(calculateSalary(thisYear, parseInt(value), selectedOption));
+    }
+  };
+
+  const handleOptionChange = (option: IOption) => {
+    setSelectedOption(option.label);
+    setSalary(calculateSalary(thisYear, examinationYear, option.label));
+  };
+
   return (
     <div className={styles.wrapper}>
       <Text type="h1">{salaryAndBenefits.basicTitle}</Text>
+      {salaryAndBenefits.showSalaryCalculator && 
+      <div className={styles.calculator}>  
+      <RadioButtonGroup
+        id="degree-group"
+        label="Choose your degree"
+        options={options}
+        onValueChange={handleOptionChange}
+      />
+      <InputField
```

Comment:
`InputField` should have `type="number"`

Reply:
and ideally a `min` and `max`, but that requires changes to `InputField`

example from outdated PR for year input:
https://github.com/varianter/variant.no/blob/1cdd430ecdf9500e995ee2c426ea870d80d61dd6/src/components/forms/inputYear/InputYear.tsx#L31

Reply:
We need to figure out if we should use inputfield, dropdown or other. We need to use whats bes regarding UU